src/main/java/com/linecorp/armeria/server/ServiceMapping.java

```diff
@@ -112,8 +112,7 @@ public MappedService apply(String path) {
         // Cache miss or disabled cache
         MappedService result = MappedService.empty();
         final int size = patterns.size();
-        for (int i = 0; i < size; i ++) {
-            final Entry<PathMapping, Service> e = patterns.get(i);
+        for (final Entry<PathMapping, Service> e : patterns) {
```

Comment:
The indexed loop here is intentional to avoid the instantiation of an Iterator. I.e. less garbage